src/main/java/com/google/api/codegen/discogapic/transformer/DiscoGapicNamer.java

```diff
@@ -95,12 +100,26 @@ public static Name methodAsName(Method method) {
     return function.join(resource);
   }
 
-  /** Return the name of the resource from a given method's path. */
-  public static Name getResourceIdentifier(Method method) {
+  /** Return the name of the qualified resource from a given method's path. */
+  public static Name getQualifiedResourceIdentifier(Method method, String qualifyingResource) {
+    String methodPath = method.flatPath();
+    Name qualifier = Name.anyCamel(qualifyingResource);
+    String baseResource = getResourceIdentifier(methodPath).toLowerCamel();
+    Name baseResourceName = Name.anyCamel(baseResource);
+
+    if (!Inflector.singularize(qualifier.toLowerCamel()).equals(baseResource)) {
+      baseResourceName = qualifier.join(baseResourceName);
+    }
```

Comment:
It doesn't seem like the output of singularization is actually used

Reply:
The output is used in checking to see if the `baseResouceName` should be qualified, but the output doesn't actually get returned in any way by this function, correct. 

`qualifyingResource` will be a plural noun. In the case that the `qualifyingResource` represents the same entity as the `baseResourceName`, it's cleaner to just use `baseResourceName`. E.g. instead of "TargetVpnGatewaysTargetVpnGatewayName", the name can just be "TargetVpnGatewayName".

Reply:
Oh! That makes sense. Maybe clarify that somewhere, because otherwise the logic looks odd.

Reply:
Added a comment!